lib/cocoapods/installer.rb

```diff
@@ -206,11 +206,12 @@ def create_generator
     #
     def generate_pods_project(generator = create_generator)
       UI.section 'Generating Pods project' do
-        @target_installation_results = generator.generate!
-        @pods_project = generator.project
+        pod_project_generation_result = generator.generate!
+        @target_installation_results = pod_project_generation_result.target_installation_results
+        @pods_project = pod_project_generation_result.project
         run_podfile_post_install_hooks
-        generator.write
-        generator.share_development_pod_schemes
+        generator.write(pods_project)
+        generator.share_development_pod_schemes(project)
```

Comment:
not sure if you want in this PR but I think those should become class methods. They contain no state

Reply:
actually they do. bleh